ash/src/extensions/khr/external_fence_fd.rs

```diff
@@ -26,10 +30,20 @@ impl ExternalFenceFd {
     }
 
     /// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkGetFenceFdKHR.html>
+    ///
+    /// May return [`None`] if the fence has already been signaled at the time this function is
+    /// called.
     #[inline]
-    pub unsafe fn get_fence_fd(&self, get_info: &vk::FenceGetFdInfoKHR) -> VkResult<i32> {
+    pub unsafe fn get_fence_fd(
+        &self,
+        get_info: &vk::FenceGetFdInfoKHR,
+    ) -> VkResult<Option<OwnedFd>> {
```

Comment:
Pushed a little update to denote that this can be `None` (when Vulkan sets it to `-1`).

The docs for the equivalent Win32 function don't state this though, perhaps there will always be a fence on Windows.

I'll take a look at the conflicts and remaining comments about polyfilling after getting some other PRs in.